Feature/research field page

euny0ung opened the pull request:
연구분야 페이지를 구현했습니다.
파란색이 진한편이라 버튼의 글자가 잘 안보이는것 같아 한단계 낮췄습니다.

![연구분야페이지 완성](https://user-images.githubusercontent.com/56199780/169819775-aa4c3c53-adca-4341-bbb5-7d68a052b6dd.PNG)

euny0ung committed:
feat : Added research field page

euny0ung committed:
feat : Added research field page

LimEunSang committed:
fix: Update ResearcherField page

- 잘못된 import 경로 수정했습니다.
- /researcherField 경로에 페이지를 렌더링하도록 코드 추가했습니다.
- module import 시 같은 module은 묶었습니다.
- 페이지 폴더로 경로 이동이 필요합니다.
- 현재 제 노트북에서 실행되지 않습니다. 오류 수정이 필요합니다.

LimEunSang committed:
chore: Add sass module dependency

- package.json 파일에 sass module 의존성 추가했습니다.
- 이제 npm i 실행하면 sass module 다운로드 됩니다.
- 추가로 경로에 대문자가 있어서 소문자로 바꿨습니다.

euny0ung committed:
refactor: Update ResearcherField
ResearcherPages 폴더 삭제 후 파일을 components로 이동

euny0ung committed:
refactor: Update ResearcherField
ResearcherPages 폴더 삭제 후 파일을 components로 이동

euny0ung committed:
refactor: Update ResearchField
더 좋은 코드로 수정하였습니다.

-Grid를 모두 Box로 변경하였습니다
-components에 있던 researchPages 폴더를 research 폴더로 옮겼습니다.
-부제목 parallax scrolling 삭제
-큰 모니터에서도 parallax scrolling되도록 수정
#15

euny0ung committed:
Merge remote-tracking branch 'origin' into feature/research_field_page

euny0ung committed:
Merge branch 'feature/research_field_page' of https://github.com/AI-Research-Center-Homepage/frontend into feature/research_field_page

euny0ung committed:
fix: Update researchField

오타를 수정하였습니다.

#15

euny0ung committed:
fix: Update ResearchField

연구분야 페이지를 전체적으로 수정하였습니다.

- 연구분야 소개글 부분을 divider에서 Box로 변경하였습니다.
- 부제목에 있던 scrolling parallax 기능을 제거하였습니다.
- 스크롤 길이가 상대적으로 나타날 수 있도록 변경하였습니다.
- 창 크기에 따라 버튼 정렬이 달라지도록 변경하였습니다.
- 글자와 버튼 색을 변경하였습니다. (연보라 -> 파랑)
- 사진이 나타나지 않는 오류를 수정하였습니다.

#15

euny0ung committed:
chore: Add comment
주석을 추가하였습니다.

#15

euny0ung commented:
css 파일은 research 폴더 안에 넣었습니다.

LimEunSang commented on src/pages/research/ResearcherFieldPages/FieldDL.js:
```diff
@@ -0,0 +1,75 @@
+      sx={{
+        width: '100%',
+        height: 'auto',
+        // backgroundColor: "#f9f9f9"
```
주석 처리한 코드를 남겨 놓은 이유가 있나요?

LimEunSang commented on src/pages/research/ResearcherFieldPages/FieldDL.js:
```diff
@@ -0,0 +1,75 @@
+        src="https://images.unsplash.com/photo-1645148100502-418e89bf99c2?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY1MjEwNzU1Mw&ixlib=rb-1.2.1&q=80&w=1080"
+        style={{
+          height: '30vw',
+          marginLeft: '70%',
```
사진을 오른쪽에 배치한 이유가 있나요?

euny0ung replied:
없습니다! 가운데에 있으면 뭔가 이상해서 오른쪽으로 빼두었습니다

LimEunSang reviewed (commented):
디자인 관련해서 신기하고 처음 보는 속성들을 많이 구경합니다..
visible에 id값을 할당해  버튼마다 다른 컴포넌트를 나오게 하는 아이디어 상당히 좋네요!
고생 많이하셨습니다!!

1. 각 분야의 컴포넌트(FildDL, FildML, ...)가 내용만 다르고 똑같은 형식을 띄고 있어서 공통 컴포넌트화하면 훨씬 효율적일 거 같아요. 각 Field 파일을 따로 생성할 필요 없이 내용만 객체 props로 전달하면 하나의 페이지 안에서 모두 선언이 가능해집니다.

2. 글자 길이가 길어서 특정 창 크기에서 저렇게 밑으로 삐져나오네요. 크기가 일정 이하 px로 떨어지면 글자 크기를 줄여서 해결할 수 있을 거 같아요
![image](https://user-images.githubusercontent.com/86942472/169873946-54aacfa7-decb-4f94-99ae-ac5d64d70909.png)

suinj8 commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,151 @@
+ *@description 연구분야 페이지
+ */
+export default function ResearchField() {
+  //    랩실 설명
```
주석 띄어쓰기가 많이 되어있네요.

suinj8 commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,151 @@
+          color: '#212121',
+          backgroundColor: 'white',
+          '&:hover': {
+            backgroundColor: '#0288d1',
```
파란색이 좀 진해서 글씨가 선명하지 않은것 같습니다.
배경색이 바뀌면서 글씨색도 흰색으로 바뀌게 하면 어떨까요?

![image](https://user-images.githubusercontent.com/90179555/169992551-43f549d9-1f81-433d-b008-d59e6dd8c470.png)

suinj8 reviewed (changes requested):
결국 그리드로 해결하셨군요.
고생하셨습니다.

euny0ung committed:
style : Update ResearchField
버튼 텍스트 색 변경
- 마우스를 올렸을 때 버튼의 텍스트 색이 흰색으로 변경되게 수정하였습니다
- themeprovider 코드가 불필요하여 제거하였습니다.

#42

euny0ung committed:
chore : Edit comment
주석 수정

#42

euny0ung committed:
refactor: Update ResearchField
-버튼 이벤트 구현방식을 바꿨습니다
-데이터를 Object로 묶어 컴포넌트 하나를 재사용하는 방식으로 바꿨습니다.

#15

euny0ung committed:
chore: add comment
주석을 추가하였습니다.

#15

euny0ung committed:
fixes #15

euny0ung committed:
chore: Add comment
주석을 추가하였습니다.

#46

euny0ung committed:
fix: change fontSize
글자 길이때문에 화면 크기가 xs로 넘어가기 직전에 기계학습 버튼 크기가 늘어지는 오류 수정
- variant 사용 -> vw 사용

euny0ung committed:
chore: fix conflict
충돌해결

#42

euny0ung committed:
style: Update ResearchField
첫번째 버튼의 화면 출력을 디폴트로 설정하였습니다.
-useState의 초기값을 trans로 변경

euny0ung commented:
말해주신 부분들 전부 수정했습니다. 확인 부탁드려요! 버튼을 눌렀을때 화면이 출력되어야하는데 됐다가 안됐다가 하는 이슈가 있습니다(이슈 46번). 이 부분 고치면 끝입니다..

euny0ung committed:
fix: Resolve problem
랜덤으로 화면이 출력되는 문제 수정
-secont -> second 오타 수정
-Typography 태그 삭제

#42

euny0ung committed:
feat: Update Header.js
비어있던 경로를 추가하였습니다.

#42

suinj8 reviewed (approved):
LGTM!

LimEunSang commented on src/App.js:
```diff
@@ -39,6 +40,7 @@ const App = () => {
           {/* 연구프로젝트 페이지 추가 예정 */}
           <Route path="/thesis" element={<Thesis />} />
           <Route path="/demo" element={<Demo />} />
+          <Route path="/researchfield" element={<ResearchField />} />
```
39번째 줄 지우고 그 자리에 작성해달라고 navigate 브랜치 PR에 글 남겨놨었는데, 신경 못쓰셨나보네요! 병합 이후에 제가 수정할 거 수정하면서 한 번에 수정하겠습니다.

euny0ung replied:
Project 파일에서 한번에 수정했는데 따로해야되는군요 주석 수정하는김에 같이 하겠습니다!!

LimEunSang commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,322 @@
+    </div>
+  );
+
+  const [contents, setContents] = useState("trans");
```
변수 명이 'field' 처럼 좀 더 명확했으면 좋았을 거 같아요. 이런 부분도 일단 병합 이후에 제가 고민해서 한 번 수정해보도록 할게요.

LimEunSang commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,322 @@
+import Footer from "../../components/Footer";
+import "./Field.scss";
+
+//임시 사진 경로
```
제가 알기로 주석처리문 뒤에 한 칸 띄고 주석 작성하는 걸로 알고있어요. 다른 분들도 그렇게 한 거 같아요.
//임시 사진 경로 (x)
// 임시 사진 경로 (o)

이 부분도 일단 병합 이후에 제가 수정할게요!

LimEunSang commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,322 @@
+const imgUrl =
+  "https://images.unsplash.com/photo-1645148100502-418e89bf99c2?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY1MjEwNzU1Mw&ixlib=rb-1.2.1&q=80&w=1080";
+
+// 버튼 클릭시 나오는 화면에 대한 정보
```
매우 사소한 거지만 앞으로 코드 작성하실 때 스타일을 통일해주세요!
정해지지 않은 방식에 대해서 상대방의 스타일에 맞춰달라는게 아니라, 예를 들어
//내용 <- 이렇게 작성하기로 마음먹었으면 전부 저렇게 작성해주시고
// 내용 <- 이렇게 작성하기로 마음먹었으면 전부 저렇게 작성해주세요!

결국 refactoring 할 때 다 맞춰야 하는데 그때 작업할 양이 많아지지 않게 하기 위해서요.

LimEunSang commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,322 @@
+      >
+        {props.title}
+      </Typography>
+      {/* 부제목 */}
```
다음에는 주석으로 구간을 명시하여 나눠줄 때 줄바꿈 해주세요!
일단 머지 이후에 제가 수정할게요!

euny0ung replied:
ResearchField 파일 주석 띄어쓰기, 구간 나누기 완료하였습니다. 구간 나누기의 경우 줄바꿈을 해도 prettier 때문에 안먹히는(?) 부분이 있어서 그런건 그냥 두었습니다. ex) ResearchField의 303번째 줄

LimEunSang commented on src/pages/research/ResearchField.js:
```diff
@@ -0,0 +1,322 @@
+      </Box>
+
+      {/* 버튼마다 다른 컴포넌트 나오게 하였음 */}
+      {contents && selectComponent[contents]}
```
이 부분은 정말 신기하네요. 난생 처음 보는 형식인데, 객체[맴버] 이렇게 접근하면 맴버에 해당하는 부분만 출력이 되는줄 처음 알았어요. 유용한거 배워갑니다!

LimEunSang reviewed (approved):
페이지 구현하느라 고생하신게 코드에서 보여요... 정말 수고많으셨어요! 함수 정의도 잘 해서 사용하고, 컴포넌트화도 너무 잘 하셨네요.  밑에 남겨놓은 comment 한 번씩만  확인해주세요! LGTM

euny0ung committed:
style: Update App.js
작은 따옴표를 큰 따옴표로 수정

#42

euny0ung committed:
refactor: Update App.js , ResearchField.js
주석 띄어쓰기 및 구간 나누기

#42

baaaam771 reviewed (approved):
기존 코드에서 중복 되는 거 제거하시고 버튼 이벤트 새로 만들어서 작성하느라 고생많으셨습니다 커밋 전체 내용 보니까 본인이 스스로 로직 생각하셔서 직접 짜신 것 같더라고요 수고하셨어요! LGTM
